app/src/main/java/com/example/gallery/FullImageFragment.kt

```diff
@@ -417,6 +422,13 @@ class FullImageFragment : Fragment() {
             var imageView = itemLayout.findViewById<ZoomageView>(R.id.fullimageViewid)
             imageView.setImageURI(Uri.parse(imageList[position]))
 
+            var currPos= if(position==imageList.size-1 || position==0)
```

Comment:
For Viewpager, currPos will have value 2. It is because viewpager will call instantiateItem() method for three times. For position 1 then for 0 and 2.  This is default behaviour of viewPager see documentation for further knowledge about offScreenPageLimit which has minimum value of 1. We can't do much here , position variable will us wrong information. 
So, try moving this logic on page selected listener. I guess, it might work.

Reply:
Yeah, Thanks 👍. But @vilgad told that he has some another logic to fixed it.

Reply:
Cool